### PR TITLE
Upgraded the dependencies

### PR DESCRIPTION
**Pull request by jamescoggan**
### Description

* Updated the dependency versions
* Disabled the WEB module for now, its not fully working

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -13,62 +16,62 @@ apply plugin: 'kotlin-android-extensions'
...
-    testCompile 'junit:junit:4.12'
-    androidTestCompile('com.android.support.test.espresso:espresso-core:3.0.1', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
+  implementation fileTree(dir: 'libs', include: ['*.jar'])
+  implementation 'com.android.support:appcompat-v7:28.0.0-beta01'
+  implementation "com.android.support:support-v4:28.0.0-beta01"
+  implementation 'com.android.support:recyclerview-v7:28.0.0-beta01'
+  implementation 'com.android.support.constraint:constraint-layout:1.1.2'
+  testImplementation 'junit:junit:4.12'
```
Comment (jarekankowski) [inline android/app/build.gradle]: For the libraries we're using across the modules, should we extract libraries versions to an external variable to keep consistency across modules and have one place where we change it (like with `kotlin_version`)?

**Comment by jarekankowski** [comment]
Also, what do you think about moving dependencies to [Kotlin + buildSrc](https://handstandsam.com/2018/02/11/kotlin-buildsrc-for-better-gradle-dependency-management/)?

File: android/app/build.gradle [hunk 2]
```diff
@@ -13,62 +16,62 @@ apply plugin: 'kotlin-android-extensions'
...
-    compile 'com.android.support:appcompat-v7:26.1.0'
-    compile 'com.android.support:recyclerview-v7:26.1.0'
-    compile 'com.android.support.constraint:constraint-layout:1.0.2'
-    testCompile 'junit:junit:4.12'
-    androidTestCompile('com.android.support.test.espresso:espresso-core:3.0.1', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
+  implementation fileTree(dir: 'libs', include: ['*.jar'])
+  implementation 'com.android.support:appcompat-v7:28.0.0-beta01'
+  implementation "com.android.support:support-v4:28.0.0-beta01"
```
[the remark this answers is not in the document]
Comment (jamescoggan) [reply android/app/build.gradle]: We could probably use `kotlinTest` also, I can add in separate PR

File: common-jvm/build.gradle [hunk 3]
```diff
@@ -2,11 +2,11 @@ apply plugin: 'kotlin-platform-jvm'
...
 
 dependencies {
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jre8:$kotlin_version"
-    compile "org.jetbrains.kotlinx:kotlinx-serialization-runtime:$serialization_version"
-    testCompile "junit:junit:4.12"
-    testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
-    testCompile "org.jetbrains.kotlin:kotlin-test:$kotlin_version"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+    implementation "org.jetbrains.kotlinx:kotlinx-serialization-runtime:$serialization_version"
+    testImplementation "junit:junit:4.12"
```
[the remark this answers is not in the document]
Comment (jamescoggan) [reply common-jvm/build.gradle]: We could probably use `kotlinTest` also, I can add in separate PR

File: build.gradle [hunk 4]
```diff
@@ -1,8 +1,8 @@
 allprojects {
     buildscript {
-        ext.kotlin_version = '1.2.0-rc-39'
-        ext.serialization_version = '0.2'
-        ext.anko_version = '0.10.1'
+        ext.kotlin_version = '1.2.51'
```
[the remark this answers is not in the document]
Comment (jamescoggan) [reply build.gradle]: I can update in separate PR, its gonna need some lint fixes

File: backend/build.gradle [hunk 5]
```diff
@@ -38,20 +38,20 @@ repositories {
...
+    implementation "io.ktor:ktor-auth:$ktor_version"
+    implementation "io.ktor:ktor-websockets:$ktor_version"
+    implementation "org.jetbrains.squash:squash:$squash_version"
+    implementation "org.jetbrains.squash:squash-h2:$squash_version"
+    implementation 'com.github.salomonbrys.kotson:kotson:2.5.0'
+    implementation "ch.qos.logback:logback-classic:1.2.1"
+    implementation group: 'com.zaxxer', name: 'HikariCP', version: '2.7.2'
 
-    testCompile group: 'junit', name: 'junit', version: '4.12'
+    testImplementation group: 'junit', name: 'junit', version: '4.12'
```
[the remark this answers is not in the document]
Comment (jamescoggan) [reply backend/build.gradle]: Sounds good for the `buildSrc` change

File: backend/build.gradle [hunk 6]
```diff
@@ -38,20 +38,20 @@ repositories {
...
+    implementation project(":common-jvm")
+    implementation "io.ktor:ktor-server-netty:$ktor_version"
+    implementation "io.ktor:ktor-gson:$ktor_version"
+    implementation "io.ktor:ktor-auth:$ktor_version"
+    implementation "io.ktor:ktor-websockets:$ktor_version"
+    implementation "org.jetbrains.squash:squash:$squash_version"
+    implementation "org.jetbrains.squash:squash-h2:$squash_version"
+    implementation 'com.github.salomonbrys.kotson:kotson:2.5.0'
+    implementation "ch.qos.logback:logback-classic:1.2.1"
+    implementation group: 'com.zaxxer', name: 'HikariCP', version: '2.7.2'
```
[the remark this answers is not in the document]
Comment (jamescoggan) [reply backend/build.gradle]: Sounds good for the `buildSrc` change

File: android/build.gradle [hunk 7]
```diff
@@ -4,10 +4,11 @@ buildscript {
     repositories {
         maven { url 'https://maven.google.com' }
         jcenter()
-        maven { url "http://dl.bintray.com/kotlin/kotlin-eap-1.2" }
+      google()
+      maven { url "http://dl.bintray.com/kotlin/kotlin-eap-1.2" }
```
[the remark this answers is not in the document]
Comment (jamescoggan) [reply android/build.gradle]: probably not, upgrade with 1.3?

File: android/app/build.gradle (hunk 1 above)
Comment (jamescoggan) [reply android/app/build.gradle]: Yeap

**Pull request merged**